### ISSUE TITLE
Crash after parsing 1959 files

### ISSUE THREAD

**Issue by crystalfp** [syntax-tree/mdast-util-from-markdown#3]
### Subject of the issue

After parsing in a loop 1959 files the `fromMarkdown()` routine crashes with the following error:
```
RangeError: Maximum call stack size exceeded
    at unravelLinkedTokens (D:\ProjectsShort\ParseMarkdown\node_modules\micromark\dist\util\subtokenize.js:122:29)
    at unravelLinkedTokens (D:\ProjectsShort\ParseMarkdown\node_modules\micromark\dist\util\subtokenize.js:158:5)
    at unravelLinkedTokens (D:\ProjectsShort\ParseMarkdown\node_modules\micromark\dist\util\subtokenize.js:158:5)
    at unravelLinkedTokens (D:\ProjectsShort\ParseMarkdown\node_modules\micromark\dist\util\subtokenize.js:158:5)
    at unravelLinkedTokens (D:\ProjectsShort\ParseMarkdown\node_modules\micromark\dist\util\subtokenize.js:158:5)
    at unravelLinkedTokens (D:\ProjectsShort\ParseMarkdown\node_modules\micromark\dist\util\subtokenize.js:158:5)
    at unravelLinkedTokens (D:\ProjectsShort\ParseMarkdown\node_modules\micromark\dist\util\subtokenize.js:158:5)
    at unravelLinkedTokens (D:\ProjectsShort\ParseMarkdown\node_modules\micromark\dist\util\subtokenize.js:158:5)
    at unravelLinkedTokens (D:\ProjectsShort\ParseMarkdown\node_modules\micromark\dist\util\subtokenize.js:158:5)
    at unravelLinkedTokens (D:\ProjectsShort\ParseMarkdown\node_modules\micromark\dist\util\subtokenize.js:158:5)
```
If I parse the very same file alone, no crash occurs.

In the loop the same crash occurs for other book-size documents, then continues without problems for other (~1000) smaller documents.

### Your environment

-    OS: Windows 10 64 bits
-    Packages: "mdast-util-from-markdown": "^0.5.0" and "@types/node": "^14.11.1"
-    Env: node: 14.8.0; npm: 6.14.7

### Steps to reproduce
I cannot provide you with all my files and a single file does not manifest the problem. Is there any way to show more info useful to debug the problem? And to monitor resource usage?

### Expected behavior

Well, that all files are parsed successfully

### Actual behavior

See above.

**Comment by wooorm** [comment]
We’re (specifically, @ChristianMurphy) doing some fuzz testing: https://github.com/micromark/micromark/pull/18.
We encountered this error before (https://github.com/micromark/micromark/issues/20) when there are say, 1000s of links in each other. Although, it seems this is pointing to a memoryleek

**Comment by crystalfp** [comment]
I think so. The files have few external URL inside and no links to anchors.

**Comment by wooorm** [comment]
This should be fixed in `micromark@2.8.1`. Could you check?

Furthermore: this issue most likely relates to micromark, not this project. So I’ll close it here. Feel free top respond here if it wasn’t fixed / open a new one on micromark!

**Comment by crystalfp** [comment]
After installing 0.6.0 it parsed 2600 files without crashing! Well, now there is something strange with `"mdast-util-frontmatter"` (it parse only the first file) but I should investigate and open to the correct place.
Thanks!
mario
